lib/domain.js

```diff
@@ -53,20 +57,22 @@ const asyncHook = createHook({
   init(asyncId, type, triggerAsyncId, resource) {
     if (process.domain !== null && process.domain !== undefined) {
       // If this operation is created while in a domain, let's mark it
-      pairing.set(asyncId, process.domain);
+      pairing.set(asyncId, process.domain[kWeak]);
```

Comment:
Is it conceivable that user code tampers with `process.domain`?

What would happen if I set `process.domain = {}`?

Reply:
I don’t think the domain code is all that tampering-proof to being with…

I’m also not sure what we could do in that case.